File: w1_p5/15.c
#include <stdio.h>
#include <math.h>

int main() {
	int n;
	double input[1000];
	
	while(scanf("%d", &n) == 1){
		if(n == 0) break;
		
		double total=0;
		double result=0;
		for(int i=0; i<n; i++){
			scanf("%lf", &input[i]);
			total += input[i];
		}
		total = total/n + 0.009;
		total = floor(total*100.0f)/100.0f;
		
		for(int i=0; i<n; i++){
			if(input[i]>total)
				result = result + input[i] - total;
		}
		
		printf("$%.2lf\n", result);
	}
	
	return 0;
}

File: w1_p5/15.java
import java.io.*;
import java.util.Scanner;
class Main {
	public static void main(String[] args) throws Exception {
		
		int n=0;

		Scanner input = new Scanner(System.in);
		while(true){
			double sum =0;
			double money=0;
			n = input.nextInt();
			
			if(n==0){
				break;
			}
			double m[] = new double[n];
			for(int i=0; i<n; i++){
				m[i]=input.nextDouble();
				sum += m[i];
			}
			sum = sum/n;
		
		for(int i=0;i<n; i++){
			if(m[i] < sum){
				money = money+(sum-m[i]);
			}
		}
		money = Math.floor(money*100)/100;
		System.out.printf("$%.2f\n",money);
	}
	}
}